### ISSUE TITLE
checklist with markdown not working properly

### ISSUE THREAD

**Issue by gocarlos** [Automattic/simplenote-electron#694]
#### Steps to reproduce
1. start a new note
2. type 
```
- [ ] asdfasdf
```
3. expected: 

4. result:

#### What I expected
- [x] asdfasfd

#### What happened instead
```
<input type="checkbox" disabled style="margin: 0px 0.35em 0.25em -1.6em; vertical-align: middle;"> asdfasdf
``` 

#### OS version
Windows 7 64 bit, using version 1.1.2

#### Screenshot / Video
![2018-01-16 21_13_23-simplenote](https://user-images.githubusercontent.com/1331762/35010311-c18cca82-fb02-11e7-861f-e0e390631b0e.png)
![2018-01-16 21_14_04-simplenote](https://user-images.githubusercontent.com/1331762/35010313-c2b6c3f4-fb02-11e7-829a-9230365a6159.png)

**Comment by gocarlos** [comment]
https://app.simplenote.com/ also does not seem to work correctly. 
http://demo.showdownjs.com/ does work correctly

**Comment by roundhill** [comment]
@dmsnell this used to work, but appears to have broken. Mind taking a peek?

**Comment by roundhill** [comment]
Oh, I bet this broke because we don't allow html tags in the preview now, they get escaped.

**Comment by mirka** [comment]
Closing as this was fixed in #724.
I've added two minor issues as enhancements: #874 #875
